### DISCUSSION TITLE
Obtaining percentile from histogram to csv

### DISCUSSION THREAD

**Discussion by RobShaw77** [ImageMagick/ImageMagick#5897] [Help]
### ImageMagick version

7.1.0-53

### Operating system, version and so on

Windows 10

### Description

I have a large number (around 130,000) single channel images (500x500 pixels). For each of these I would like to get the 10th percentile (i.e. the shade of the 25000th darkest pixel) and write the values out to text file. 

Looping through all the images should be straightforward but I can' find a way to get IM to output a single figure rather than the whole histogram.

Can anyone help please?

**Comment by snibgo** [comment]
A simple solution: rearrange the rows of pixels to be in a single row, sort the pixels, and examine pixel number 25000:
```
magick in.png -crop x1 +repage +append +repage -sort-pixels -crop 1x1+2000+0+0 +repage txt:
```
Sadly, the built-in IM sort is pathetically slow (100 seconds for 500x500). My process module "sortpixels" is much faster (0.5 seconds for 500x500). See [Process modules: sort pixels](http://im.snibgo.com/customim.htm#sortpixels)
```
magick in.png -crop x1 +repage +append +repage -process sortpixels -crop 1x1+25000+0+0 +repage txt:
```
A more complex solution that doesn't need a sort is with cumulative histograms. Sadly, my brain isn't working today, but [Process modules: mkhisto, make histogram](http://im.snibgo.com/customim.htm#mkhisto) might help.

If you build IM from source, adding my process modules is quite simple.
